### PR TITLE
2571 enhance decollate arg of workflows

### PR DESCRIPTION
**Pull request by Nic-Ma**
Part of #2571 

### Description
This PR enhanced the `decollate` arg of workflows to provide more flexible usage.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/engines/evaluator.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ class Evaluator(Workflow):
             for more details, check: https://pytorch.org/ignite/generated/ignite.engine.engine.Engine.html
             #ignite.engine.engine.Engine.register_events.
         decollate: whether to decollate the batch-first data to a list of data after model computation,
-            default to `True`. if `False`, postprocessing will be ignored as the `monai.transforms` module
-            assumes channel-first data.
+            default to `True`. if `False`, as the `monai.transforms` module assumes channel-first data,
+            please don't set `monai.transforms` for `postprocessing`.
```
Comment (wyli) [inline monai/engines/evaluator.py:71]: how about:
```suggestion
            default to `True`. We recommend `decollate=True` when `postprocessing` uses components from 
           `monai.transforms`.
```

**Review by wyli — APPROVED** [review]
thanks, the docstrings should make a recommendation, otherwise it looks good

File: monai/engines/evaluator.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/engines/evaluator.py]: Sure, updated in the latest commit.
Thanks.

**Pull request merged**